bluepysnap/network.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2020-2021, EPFL/Blue Brain Project
+
+# This file is part of BlueBrain SNAP library <https://github.com/BlueBrain/snap>
+
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""Module containing the Abstract classes for the Network."""
+import abc
+
+import pandas as pd
+import numpy as np
+from cached_property import cached_property
+
+
+from bluepysnap import utils
+from bluepysnap.exceptions import BluepySnapError
+
+
+class NetworkObject(abc.ABC):
+    """Abstract class for the top level NetworkObjects accessor."""
+
+    def __init__(self, circuit):
+        """Initialize the top level NetworkObjects accessor."""
+        self._circuit = circuit
+
+    def _get_populations(self, cls, config):
+        """Collects the different NetworkObjectPopulation and returns them as a dict."""
+        res = {}
+        for file_config in config:
+            storage = cls(file_config, self._circuit)
+            for population in storage.population_names:  # pylint: disable=not-an-iterable
+                if population in res:
+                    raise BluepySnapError(
+                        "Duplicated {} population: '{}'".format(self.__class__.__name__,
+                                                                population))
+                res[population] = storage.population(population)
+        return res
+
+    @abc.abstractmethod
+    def _collect_populations(self):
+        """Should specify the self._get_populations arguments."""
+
+    @cached_property
+    def _config(self):
+        return self._circuit.config
+
+    @cached_property
+    def _populations(self):
+        """Cached population dictionary."""
+        return self._collect_populations()
+
+    @cached_property
+    def population_names(self):
+        """Returns all the NetworkObjects population names from the Circuit."""
```

Comment:
You can add the fact it returns a sorted object.

Reply:
done